base/src/com/google/idea/blaze/base/sync/data/DelegatingBlazeProjectDataManager.java

```diff
@@ -61,6 +54,27 @@ public BlazeProjectData loadProject(BlazeImportSettings importSettings) {
 
   @Override
   public void saveProject(BlazeImportSettings importSettings, BlazeProjectData projectData) {
+    if (projectType != Blaze.getProjectType(project)) {
+      initBlazeProjectDataDelegate(project);
```

Comment:
Probably we need to call `save` here for the old model... But this is a thing only for Aspect one.

Reply:
With 50% of non-test impls not using `saveProject` this interface definitely needs some cleanup...
```    
// TODO(mathewi) Tidy up the interface to remove this unnecessary stuff.
```